libr/anal/fcn.c

```diff
@@ -1818,14 +1820,38 @@ R_API int r_anal_fcn_add_bb(RAnal *anal, RAnalFunction *fcn, ut64 addr, ut64 siz
 			r_anal_fcn_update_tinyrange_bbs (fcn);
 		}
 	}
-	if (!bb) {
-		bb = appendBasicBlock (anal, fcn, addr);
+	if (x86) {
+		if (bb) {
+			r_list_delete_data (fcn->bbs, bb);
+		}
+		ut8 *bbuf = malloc (size);
+		if (bbuf) {
+			anal->iob.read_at (anal->iob.io, addr, bbuf, size);
+			fcn_recurse (anal, fcn, addr, bbuf, size, 1);
+			r_anal_fcn_update_tinyrange_bbs (fcn);
+			free (bbuf);
+		}
+		bb = NULL;
+		r_list_foreach (fcn->bbs, iter, bbi) {
```

Comment:
use `r_anal_fcn_bbget_at`. it should be what you are looking for.

Reply:
ok 4a25925a379ffc01c94504dd733df6dddd532348